### ISSUE TITLE
[Feature Request]: as.teal_slice extension for filter_expr

### ISSUE THREAD

**Issue by m7pr** [insightsengineering/teal.slice#348]
### Feature description

Currently on `filter_panel_refactor@main` `as.teal_slice` uses `filter_var` to create a `teal_slice` object.
Would it be helpful if we also allow to call `filter_expr` based on the names of the object that is included in `as.teal_slice`, in case it's a list with parameter specification for `filter_expr` call?

https://github.com/insightsengineering/teal.slice/blob/724dd0d8e9ce4dee76a3127af91b8dd5014e4227/R/teal_slice.R#L250-L253

Something like

```{R}
  fun <- if ("expr" %in% names(x)) {
    filter_expr
  } else {
    filter_var
  }

  do.call(fun, x)
```

This could be helpful in the restoring of teal_slices https://github.com/insightsengineering/teal.slice/pull/339

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

**Comment by chlebowa** [comment]
I think I would prefer a separate function `as.teal_slices_expr`.

**Comment by m7pr** [comment]
Here we saw https://github.com/insightsengineering/teal.slice/pull/339#issuecomment-1604027386 that it's more convenient to have one `as.teal_slice` function

**Comment by m7pr** [comment]
potentially gonna be merged with #339
